Adds PPSSPP and Citron emulators and configs into base-emu image also fixes download issue for XEMU and Dolphin AppImages

Hotcooler opened the pull request:
As seen in the title, also adds relevant config files that provide basic configuration (like sound and paths) out of the box. Also adds launch commands for Pegasus.

Hotcooler commented on apps/pegasus/build/configs/ppsspp/controls.ini:
```diff
@@ -0,0 +1,23 @@
+﻿[ControlMapping]
+Up = 1-19,10-19
+Down = 1-20,10-20
+Left = 1-21,10-21
+Right = 1-22,10-22
+Circle = 1-52,10-190
+Cross = 1-54,10-189
+Square = 1-29,10-191
+Triangle = 1-47,10-188
+Start = 1-62,10-197
+Select = 1-66,10-196
+L = 1-45,10-194
+R = 1-51,10-195
+An.Up = 1-37,10-4003
+An.Down = 1-39,10-4002
+An.Left = 1-38,10-4001
+An.Right = 1-40,10-4000
+Analog limiter = 1-60
+RapidFire = 1-59
+Fast-forward = 1-61
+Pause = 1-111,10-109,10-104
+SpeedToggle = 1-68
+Rewind = 1-67
```
Default bindings, first is some sort of keyboard keycode, second is a gamepad. We can probably not bundle controller bindings.

ABeltramo replied:
Same as on the other comment then, we should try and see if it's able to pick up a joypad without any hardcoded binding

Hotcooler commented on apps/pegasus/build/configs/citron/qt-config.ini:
```diff
@@ -0,0 +1,1453 @@
+player_1_button_l\default=false
+player_1_button_l="toggle:0,code:81,engine:keyboard"
+player_1_button_r\default=false
+player_1_button_r="toggle:0,code:69,engine:keyboard"
+player_1_button_zl\default=false
+player_1_button_zl="toggle:0,code:82,engine:keyboard"
+player_1_button_zr\default=false
+player_1_button_zr="toggle:0,code:84,engine:keyboard"
+player_1_button_plus\default=false
+player_1_button_plus="toggle:0,code:77,engine:keyboard"
+player_1_button_minus\default=false
+player_1_button_minus="toggle:0,code:78,engine:keyboard"
+player_1_button_dleft\default=false
+player_1_button_dleft="toggle:0,code:16777234,engine:keyboard"
+player_1_button_dup\default=false
+player_1_button_dup="toggle:0,code:16777235,engine:keyboard"
+player_1_button_dright\default=false
+player_1_button_dright="toggle:0,code:16777236,engine:keyboard"
+player_1_button_ddown\default=false
+player_1_button_ddown="toggle:0,code:16777237,engine:keyboard"
+player_1_button_slleft\default=false
+player_1_button_slleft="toggle:0,code:81,engine:keyboard"
+player_1_button_srleft\default=false
+player_1_button_srleft="toggle:0,code:69,engine:keyboard"
+player_1_button_home\default=false
+player_1_button_home="toggle:0,code:0,engine:keyboard"
+player_1_button_screenshot\default=false
+player_1_button_screenshot="toggle:0,code:0,engine:keyboard"
+player_1_button_slright\default=false
+player_1_button_slright="toggle:0,code:81,engine:keyboard"
+player_1_button_srright\default=false
+player_1_button_srright="toggle:0,code:69,engine:keyboard"
+player_1_lstick\default=false
+player_1_lstick="modifier_scale:0.500000,modifier:toggle$00$1code$016777248$1engine$0keyboard,right:toggle$00$1code$068$1engine$0keyboard,left:toggle$00$1code$065$1engine$0keyboard,down:toggle$00$1code$083$1engine$0keyboard,up:toggle$00$1code$087$1engine$0keyboard,engine:analog_from_button"
+player_1_rstick\default=false
+player_1_rstick="modifier_scale:0.500000,modifier:toggle$00$1code$00$1engine$0keyboard,right:toggle$00$1code$076$1engine$0keyboard,left:toggle$00$1code$074$1engine$0keyboard,down:toggle$00$1code$075$1engine$0keyboard,up:toggle$00$1code$073$1engine$0keyboard,engine:analog_from_button"
+player_1_motionleft\default=false
+player_1_motionleft="toggle:0,code:55,engine:keyboard"
+player_1_motionright\default=false
+player_1_motionright="toggle:0,code:56,engine:keyboard"
```
It defaults to keyboard, it's just a pre-set  for Xbox One joypad. Thought it might be a good idea to have a starting config with the likely most popular controller.  It has hotplug and SDL and built in presets for at least switch pro controller and Xbox, probably others as well. Basically all it currently does, on first boot if you have xbox controller connected, you wont have to go setup controls. But as with PPSSPP we can probably skip bundling controller bindings.

ABeltramo replied:
I'd try to remove all those hardcoded bindings then and see if it'll automatically pick up a connected Joypad. If it uses SDL under the hood, that should already work out of the box without any hardcoded config.  

I can try it on my side too with a DualSense joypad if all you have available is a xbox controller.

Hotcooler commented:
Updated with removed control mappings.

ABeltramo commented:
Have you tested it with a joypad? Does it work out of the box or does it need manual configuration?

Hotcooler commented:
> Have you tested it with a joypad? Does it work out of the box or does it need manual configuration?

Yeah both autoconfigure when controller is detected.

Hotcooler commented:
Updated the request with some more changes. 

- Added nemo file manager and it's associated icon to waybar in base-app. (mostly needed for show me cheats/patches/other folder functionality in emulators and elsewhere"
- Added mangohud and default config with it hidden to base-app
- Added ShadPS4 to base-emu (it can indeed run Bloodborn well, though sound needs checking, since it's missing at the moment)
- Changed download logic in base-emu to support tags and restored the "exclude" packages functionality to working state.
- Changed launch scripts in base emu to launch with mangohud
- Associated changes to heroic to not install mangohud again
- Adds config changes to Pegasus to launch new emulators

just-for-death commented:
how can i use shad or citron or eden in wolf
